From f60747d001c7ad47a4ac51ce13e60a236ac1b807 Mon Sep 17 00:00:00 2001
From: Dong Chenchen <dongchenchen2@huawei.com>
Date: Mon, 5 Jun 2023 21:12:21 +0800
Subject: [PATCH 259/263] [Huawei] io_uring: fix KABI broken in struct
 proto_ops

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188217

--------------------------------

commit:a3025359ffa7("net: remove cmsg restriction from io_uring based
send/recvmsg calls") removed unused flag in proto_ops, which caused
kabi issue. Revert it partially to fix it.

Signed-off-by: Dong Chenchen <dongchenchen2@huawei.com>
---
 include/linux/net.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/net.h b/include/linux/net.h
index 654abfa39f80..378a3309cd47 100644
--- a/include/linux/net.h
+++ b/include/linux/net.h
@@ -137,6 +137,7 @@ typedef int (*sk_read_actor_t)(read_descriptor_t *, struct sk_buff *,
 
 struct proto_ops {
 	int		family;
+	unsigned int    flags;
 	struct module	*owner;
 	int		(*release)   (struct socket *sock);
 	int		(*bind)	     (struct socket *sock,
-- 
2.17.1

